src/Engine/ProtoCore/DSASM/InstructionSet.cs

```diff
@@ -451,23 +683,35 @@ public static StackValue BuildArrayKey(int arrayPtr, int index)
         {
             StackValue value = new StackValue();
             value.optype = AddressType.ArrayKey;
-            value.opdata = index;
-            value.opdata_d = arrayPtr;
+
+            if (index == Constants.kInvalidIndex || arrayPtr == Constants.kInvalidIndex)
+            {
+                value.opdata = Constants.kInvalidIndex;
+            }
+            else
+            {
+                // Array key information is encoded in 64bits opdata. 
+                //
+                // High 32 bits: array pointer
+                // Low 32 bits : array key
```

Comment:
Maybe the description can be enhanced, but before that, let me guess... _array pointer_ means the starting address of the array, and _array key_ is the index into that array?